### ISSUE TITLE
[JENKINS-43022] Not correct exclude path to domains

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/git-client-plugin#1541]
<p>I am using the Credential plugin for Jenkins, and I think I have detected a bug.</p>

<p>I am going to explain the steps I have followed .(sorry for my English <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<ol>
	<li>First, I create the domain and exclude the path XXXX<br/>
  <span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/36716/36716_image1.png" width="800" style="border: 0px solid black" /></span> </li>
	<li>Second, I created the credentials for my user inside this domain<br/>
  <span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/36717/36717_image2.png" width="800" style="border: 0px solid black" /></span> </li>
	<li>Then, I configure the job, pasting the url repo.<br/>
   <span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/36715/36715_image3.png" width="800" style="border: 0px solid black" /></span> </li>
	<li>And the credentials show me jsanchra and None values. I guess the Credentials only should show None value, is that correct? Maybe I have a configuration mistake?</li>
</ol>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jsanchra.png" title="jsanchra's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jsanchra">jsanchra</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022" target="_blank">Not correct exclude path to domains</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite">markewaite</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Blocker
<li><b>component(s)</b>: git-client-plugin
<li><b>label(s)</b>: Plugin, credentials, git, jenkins
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2017-07-02T22:31:48+00:00
<li><b>votes</b>: 15
<li><b>watchers</b>: 8
<li><b>imported</b>: 20251211-071809
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I am using the Credential plugin for Jenkins, and I think I have detected a bug.</p>

<p>I am going to explain the steps I have followed .(sorry for my English <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<ol>
	<li>First, I create the domain and exclude the path XXXX
  <span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/36716/36716_image1.png" width="800" style="border: 0px solid black" /></span> </li>
	<li>Second, I created the credentials for my user inside this domain
  <span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/36717/36717_image2.png" width="800" style="border: 0px solid black" /></span> </li>
	<li>Then, I configure the job, pasting the url repo.
   <span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/36715/36715_image3.png" width="800" style="border: 0px solid black" /></span> </li>
	<li>And the credentials show me jsanchra and None values. I guess the Credentials only should show None value, is that correct? Maybe I have a configuration mistake?</li>
</ol>
</pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Jenkins v1.625.3</code></li></ul>
<details><summary><i>11 attachments</i></summary>

- [credential-exclusion-a-01.PNG](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/38146/credential-exclusion-a-01.PNG)
  > ![credential-exclusion-a-01.PNG](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/38146/credential-exclusion-a-01.PNG)
- [credential-exclusion-b-01.PNG](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/38148/credential-exclusion-b-01.PNG)
  > ![credential-exclusion-b-01.PNG](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/38148/credential-exclusion-b-01.PNG)
- [credentials-exclusion-a-02.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/38147/credentials-exclusion-a-02.png)
  > ![credentials-exclusion-a-02.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/38147/credentials-exclusion-a-02.png)
- [credentials-exclusion-a-03.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/38150/credentials-exclusion-a-03.png)
  > ![credentials-exclusion-a-03.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/38150/credentials-exclusion-a-03.png)
- [credentials-exclusion-b-02.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/38152/credentials-exclusion-b-02.png)
  > ![credentials-exclusion-b-02.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/38152/credentials-exclusion-b-02.png)
- [credentials-exclusion-b-03.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/38151/credentials-exclusion-b-03.png)
  > ![credentials-exclusion-b-03.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/38151/credentials-exclusion-b-03.png)
- [image1.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/36716/image1.png)
  > ![image1.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/36716/image1.png)
- [image2.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/36717/image2.png)
  > ![image2.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/36717/image2.png)
- [image3.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/36715/image3.png)
  > ![image3.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/36715/image3.png)
- [image5.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/36719/image5.png)
  > ![image5.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/36719/image5.png)
- [image6.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/36720/image6.png)
  > ![image6.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/36720/image6.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/stephenconnolly.png" title="stephenconnolly's avatar" /> stephenconnolly:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=292802&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-292802">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>So you have not supplied a wildcard on the path, which means that you are excluding only the EXACT path <tt>/git/STEPS/</tt> try changing your exclude path to <tt>/git/STEPS/**</tt></p>
</pre>
</details>

</ul>
</details>

<p>So you have not supplied a wildcard on the path, which means that you are excluding only the EXACT path <tt>/git/STEPS/</tt> try changing your exclude path to <tt>/git/STEPS/**</tt></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/stephenconnolly.png" title="stephenconnolly's avatar" /> stephenconnolly:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=292803&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-292803">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Marking as not a defect as I suspect it is a configuration error on your side. Please re-open if my suggestion does not resolve the issue for you. Please close if my suggestion did fix</p>
</pre>
</details>

</ul>
</details>

<p>Marking as not a defect as I suspect it is a configuration error on your side. Please re-open if my suggestion does not resolve the issue for you. Please close if my suggestion did fix</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jsanchra.png" title="jsanchra's avatar" /> jsanchra:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=292805&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-292805">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi stephen</p>

<p>Change exclude to <b>/git/STEPS/</b>** in the same problem :</p>

<p> </p>
<ul>
	<li>image 5</li>
</ul>


<p>and show credentials :</p>
<ul>
	<li>image 6</li>
</ul>


<p> </p>

<p>Thanks</p>
</pre>
</details>

</ul>
</details>

<p>Hi stephen</p>

<p>Change exclude to <b>/git/STEPS/</b>** in the same problem :</p>

<p> </p>
<ul>
	<li>image 5</li>
</ul>


<p>and show credentials :</p>
<ul>
	<li>image 6</li>
</ul>


<p> </p>

<p>Thanks</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/stephenconnolly.png" title="stephenconnolly's avatar" /> stephenconnolly:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=292808&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-292808">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is a bug in the git-client plugin.</p>

<p><a href="https://github.com/jenkinsci/git-client-plugin/blob/master/src/main/java/org/jenkinsci/plugins/gitclient/GitURIRequirementsBuilder.java#L189" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-client-plugin/blob/master/src/main/java/org/jenkinsci/plugins/gitclient/GitURIRequirementsBuilder.java#L189</a></p>

<p>Does not generate PathRequirement information at all</p>
</pre>
</details>

</ul>
</details>

<p>This is a bug in the git-client plugin.</p>

<p><a href="https://github.com/jenkinsci/git-client-plugin/blob/master/src/main/java/org/jenkinsci/plugins/gitclient/GitURIRequirementsBuilder.java#L189" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-client-plugin/blob/master/src/main/java/org/jenkinsci/plugins/gitclient/GitURIRequirementsBuilder.java#L189</a></p>

<p>Does not generate PathRequirement information at all</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/stephenconnolly.png" title="stephenconnolly's avatar" /> stephenconnolly:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=292809&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-292809">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>NOTE: The credentials plugin's helper for generating requirements does supply the path requirement: <a href="https://github.com/jenkinsci/credentials-plugin/blob/master/src/main/java/com/cloudbees/plugins/credentials/domains/URIRequirementBuilder.java#L101" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/credentials-plugin/blob/master/src/main/java/com/cloudbees/plugins/credentials/domains/URIRequirementBuilder.java#L101</a></p>
</pre>
</details>

</ul>
</details>

<p>NOTE: The credentials plugin's helper for generating requirements does supply the path requirement: <a href="https://github.com/jenkinsci/credentials-plugin/blob/master/src/main/java/com/cloudbees/plugins/credentials/domains/URIRequirementBuilder.java#L101" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/credentials-plugin/blob/master/src/main/java/com/cloudbees/plugins/credentials/domains/URIRequirementBuilder.java#L101</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jsanchra.png" title="jsanchra's avatar" /> jsanchra:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=293441&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-293441">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>something new? <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=stephenconnolly" class="user-hover" rel="stephenconnolly">stephenconnolly</a> or <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> ? .</p>

<p> </p>

<p>Thanks</p>
</pre>
</details>

</ul>
</details>

<p>something new? <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=stephenconnolly" class="user-hover" rel="stephenconnolly">stephenconnolly</a> or <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> ? .</p>

<p> </p>

<p>Thanks</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=293445&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-293445">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Nothing new.  Other items are currently higher priority.  I'd love to have additional help with the plugin.  If you're able to propose a pull request for the plugin, it tends to proceed more quickly than if I do it.</p>
</pre>
</details>

</ul>
</details>

<p>Nothing new.  Other items are currently higher priority.  I'd love to have additional help with the plugin.  If you're able to propose a pull request for the plugin, it tends to proceed more quickly than if I do it.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jsanchra.png" title="jsanchra's avatar" /> jsanchra:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=293446&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-293446">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>ok <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> , the correct url to pull request is <a href="https://github.com/jenkinsci/credentials-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/credentials-plugin</a> or <a href="https://github.com/jenkinsci/git-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin</a> ?</p>

<p>Thanks</p>

<p>p.d: We will seek all possible ways</p>

<p> </p>
</pre>
</details>

</ul>
</details>

<p>ok <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> , the correct url to pull request is <a href="https://github.com/jenkinsci/credentials-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/credentials-plugin</a> or <a href="https://github.com/jenkinsci/git-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin</a> ?</p>

<p>Thanks</p>

<p>p.d: We will seek all possible ways</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=293447&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-293447">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The <a href="https://github.com/jenkinsci/git-client-plugin" class="external-link" target="_blank" rel="nofollow noopener">git client plugin</a> is likely where the change will be needed.</p>
</pre>
</details>

</ul>
</details>

<p>The <a href="https://github.com/jenkinsci/git-client-plugin" class="external-link" target="_blank" rel="nofollow noopener">git client plugin</a> is likely where the change will be needed.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jsanchra.png" title="jsanchra's avatar" /> jsanchra:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=294026&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-294026">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hello <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a></p>

<p>Many people see this interest in this problem, can be prioritized ?.</p>

<p>Thank you</p>
</pre>
</details>

</ul>
</details>

<p>Hello <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a></p>

<p>Many people see this interest in this problem, can be prioritized ?.</p>

<p>Thank you</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=294037&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-294037">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It is assigned to me.  My time to look at bugs is limited while I search for employment (recently laid off by my current employer).  This bug and the two submodule configuration bugs are the top of my current list.</p>
</pre>
</details>

</ul>
</details>

<p>It is assigned to me.  My time to look at bugs is limited while I search for employment (recently laid off by my current employer).  This bug and the two submodule configuration bugs are the top of my current list.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jsanchra.png" title="jsanchra's avatar" /> jsanchra:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=294039&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-294039">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Ok thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> ! <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>Ok thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> ! <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jsanchra.png" title="jsanchra's avatar" /> jsanchra:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=296379&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296379">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi, <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> , something new?</p>

<p>thanks</p>
</pre>
</details>

</ul>
</details>

<p>Hi, <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> , something new?</p>

<p>thanks</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=296400&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296400">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Nothing new to report.  I'm preparing releases of the git client plugin and git plugin which will not include any change for this.  Once that is done, I'll investigate the submodule issues and this bug.</p>
</pre>
</details>

</ul>
</details>

<p>Nothing new to report.  I'm preparing releases of the git client plugin and git plugin which will not include any change for this.  Once that is done, I'll investigate the submodule issues and this bug.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jsanchra.png" title="jsanchra's avatar" /> jsanchra:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=297962&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297962">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a></p>

<p>Any news about the incidence?</p>

<p>I am looking to help you to speed up this problem, could you give me some reference of which classes to modify ?.</p>

<p>Thank you</p>
</pre>
</details>

</ul>
</details>

<p>Hi <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a></p>

<p>Any news about the incidence?</p>

<p>I am looking to help you to speed up this problem, could you give me some reference of which classes to modify ?.</p>

<p>Thank you</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=298028&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-298028">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I've nothing new to report.  I suspect the change needs a review of all locations in the git client plugin which refer to a credential, including the user interface portions and  CliGitAPIImpl and JGitAPIImpl.</p>
</pre>
</details>

</ul>
</details>

<p>I've nothing new to report.  I suspect the change needs a review of all locations in the git client plugin which refer to a credential, including the user interface portions and  CliGitAPIImpl and JGitAPIImpl.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jsanchra.png" title="jsanchra's avatar" /> jsanchra:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=298798&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-298798">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> this solution to pull request : <a href="https://github.com/jenkinsci/git-client-plugin/pull/242" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-client-plugin/pull/242</a> , tested my local and good working.</p>

<p>Please you feedback to solution.</p>

<p>Thanks</p>
</pre>
</details>

</ul>
</details>

<p>Hi <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> this solution to pull request : <a href="https://github.com/jenkinsci/git-client-plugin/pull/242" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-client-plugin/pull/242</a> , tested my local and good working.</p>

<p>Please you feedback to solution.</p>

<p>Thanks</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=301056&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-301056">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jsanchra" class="user-hover" rel="jsanchra">jsanchra</a> I've tested the pull request and it doesn't behave the way I'd expected.</p>

<p>I duplicated the bug by defining a GitHub credential in the <a href="https://github.com/MarkEWaite/docker-lfs/tree/lts-with-plugins" class="external-link" target="_blank" rel="nofollow noopener">lts-with-plugins branch of my docker instance</a> with hostname <tt>github.com</tt>, URI scheme <tt>https</tt>, and URI path exclusion <tt>/MarkEWaite/jenkins-bugs-private*</tt> . When I open the job definition and select the credentials dropdown, it (incorrectly) shows the excluded credential, in addition to the matching credentials.  That's the bug.</p>

<p>When I merged your pull request onto a <a href="https://github.com/MarkEWaite/git-client-plugin/tree/master-PR242-fix-credentials-matching" class="external-link" target="_blank" rel="nofollow noopener">development branch</a> based on the master branch with your changes, when I open the job definition and select the credentials dropdown, it (incorrectly) shows the excluded credential, in addition to the matching credentials. That's good.</p>

<p>Unfortunately, when I configure a job which should still show the credential, it does not show the credential. It is as though your changes have caused that credential to be hidden in all cases.</p>
<h3><a name="BeforeyourChanges"></a>Before your Changes</h3>

<p>Define the credential: 
 <span class="image-wrap" style=""><a id="38146_thumb" href="https://issues.jenkins.io/secure/attachment/38146/38146_credential-exclusion-a-01.PNG" title="credential-exclusion-a-01.PNG" file-preview-type="image" file-preview-id="38146" file-preview-title="credential-exclusion-a-01.PNG"><img src="https://issues.jenkins.io/secure/thumbnail/38146/_thumb_38146.png" style="border: 0px solid black" role="presentation"/></a></span></p>

<p>See the credential (unexpected): 
 <span class="image-wrap" style=""><a id="38147_thumb" href="https://issues.jenkins.io/secure/attachment/38147/38147_credentials-exclusion-a-02.png" title="credentials-exclusion-a-02.png" file-preview-type="image" file-preview-id="38147" file-preview-title="credentials-exclusion-a-02.png"><img src="https://issues.jenkins.io/secure/thumbnail/38147/_thumb_38147.png" style="border: 0px solid black" role="presentation"/></a></span></p>

<p>See the credential (expected) 
 <span class="image-wrap" style=""><a id="38150_thumb" href="https://issues.jenkins.io/secure/attachment/38150/38150_credentials-exclusion-a-03.png" title="credentials-exclusion-a-03.png" file-preview-type="image" file-preview-id="38150" file-preview-title="credentials-exclusion-a-03.png"><img src="https://issues.jenkins.io/secure/thumbnail/38150/_thumb_38150.png" style="border: 0px solid black" role="presentation"/></a></span></p>
<h3><a name="AfteryourChanges"></a>After your Changes</h3>

<p>Define the credential: 
 <span class="image-wrap" style=""><a id="38148_thumb" href="https://issues.jenkins.io/secure/attachment/38148/38148_credential-exclusion-b-01.PNG" title="credential-exclusion-b-01.PNG" file-preview-type="image" file-preview-id="38148" file-preview-title="credential-exclusion-b-01.PNG"><img src="https://issues.jenkins.io/secure/thumbnail/38148/_thumb_38148.png" style="border: 0px solid black" role="presentation"/></a></span></p>

<p>Don't see the credential (as expected) 
 <span class="image-wrap" style=""><a id="38152_thumb" href="https://issues.jenkins.io/secure/attachment/38152/38152_credentials-exclusion-b-02.png" title="credentials-exclusion-b-02.png" file-preview-type="image" file-preview-id="38152" file-preview-title="credentials-exclusion-b-02.png"><img src="https://issues.jenkins.io/secure/thumbnail/38152/_thumb_38152.png" style="border: 0px solid black" role="presentation"/></a></span></p>

<p>Don't see the credential (unexpected) 
 <span class="image-wrap" style=""><a id="38151_thumb" href="https://issues.jenkins.io/secure/attachment/38151/38151_credentials-exclusion-b-03.png" title="credentials-exclusion-b-03.png" file-preview-type="image" file-preview-id="38151" file-preview-title="credentials-exclusion-b-03.png"><img src="https://issues.jenkins.io/secure/thumbnail/38151/_thumb_38151.png" style="border: 0px solid black" role="presentation"/></a></span></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jsanchra" class="user-hover" rel="jsanchra">jsanchra</a> I've tested the pull request and it doesn't behave the way I'd expected.</p>

<p>I duplicated the bug by defining a GitHub credential in the <a href="https://github.com/MarkEWaite/docker-lfs/tree/lts-with-plugins" class="external-link" target="_blank" rel="nofollow noopener">lts-with-plugins branch of my docker instance</a> with hostname <tt>github.com</tt>, URI scheme <tt>https</tt>, and URI path exclusion <tt>/MarkEWaite/jenkins-bugs-private*</tt> . When I open the job definition and select the credentials dropdown, it (incorrectly) shows the excluded credential, in addition to the matching credentials.  That's the bug.</p>

<p>When I merged your pull request onto a <a href="https://github.com/MarkEWaite/git-client-plugin/tree/master-PR242-fix-credentials-matching" class="external-link" target="_blank" rel="nofollow noopener">development branch</a> based on the master branch with your changes, when I open the job definition and select the credentials dropdown, it (incorrectly) shows the excluded credential, in addition to the matching credentials. That's good.</p>

<p>Unfortunately, when I configure a job which should still show the credential, it does not show the credential. It is as though your changes have caused that credential to be hidden in all cases.</p>
<h3><a name="BeforeyourChanges"></a>Before your Changes</h3>

<p>Define the credential: <br/>
 <span class="image-wrap" style=""><a id="38146_thumb" href="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/38146/credential-exclusion-a-01.PNG" title="credential-exclusion-a-01.PNG" file-preview-type="image" file-preview-id="38146" file-preview-title="credential-exclusion-a-01.PNG"><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/38146/credential-exclusion-a-01.PNG" style="border: 0px solid black" role="presentation"/></a></span></p>

<p>See the credential (unexpected): <br/>
 <span class="image-wrap" style=""><a id="38147_thumb" href="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/38147/credentials-exclusion-a-02.png" title="credentials-exclusion-a-02.png" file-preview-type="image" file-preview-id="38147" file-preview-title="credentials-exclusion-a-02.png"><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/38147/credentials-exclusion-a-02.png" style="border: 0px solid black" role="presentation"/></a></span></p>

<p>See the credential (expected) <br/>
 <span class="image-wrap" style=""><a id="38150_thumb" href="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/38150/credentials-exclusion-a-03.png" title="credentials-exclusion-a-03.png" file-preview-type="image" file-preview-id="38150" file-preview-title="credentials-exclusion-a-03.png"><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/38150/credentials-exclusion-a-03.png" style="border: 0px solid black" role="presentation"/></a></span></p>
<h3><a name="AfteryourChanges"></a>After your Changes</h3>

<p>Define the credential: <br/>
 <span class="image-wrap" style=""><a id="38148_thumb" href="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/38148/credential-exclusion-b-01.PNG" title="credential-exclusion-b-01.PNG" file-preview-type="image" file-preview-id="38148" file-preview-title="credential-exclusion-b-01.PNG"><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/38148/credential-exclusion-b-01.PNG" style="border: 0px solid black" role="presentation"/></a></span></p>

<p>Don't see the credential (as expected) <br/>
 <span class="image-wrap" style=""><a id="38152_thumb" href="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/38152/credentials-exclusion-b-02.png" title="credentials-exclusion-b-02.png" file-preview-type="image" file-preview-id="38152" file-preview-title="credentials-exclusion-b-02.png"><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/38152/credentials-exclusion-b-02.png" style="border: 0px solid black" role="presentation"/></a></span></p>

<p>Don't see the credential (unexpected) <br/>
 <span class="image-wrap" style=""><a id="38151_thumb" href="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/38151/credentials-exclusion-b-03.png" title="credentials-exclusion-b-03.png" file-preview-type="image" file-preview-id="38151" file-preview-title="credentials-exclusion-b-03.png"><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/38151/credentials-exclusion-b-03.png" style="border: 0px solid black" role="presentation"/></a></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jsanchra.png" title="jsanchra's avatar" /> jsanchra:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=301115&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-301115">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> the problem es .git, ok i review this case.</p>

<p>Thanks</p>
</pre>
</details>

</ul>
</details>

<p>Hi <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> the problem es .git, ok i review this case.</p>

<p>Thanks</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=305439&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-305439">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jsanchra" class="user-hover" rel="jsanchra">jsanchra</a> I revised your pull request to reduce differences to the upstream master, and to add more tests, and have submitted it as a <a href="https://github.com/jenkinsci/git-client-plugin/pull/249" class="external-link" target="_blank" rel="nofollow noopener">new pull request</a>.  It has passed my interactive testing.  I've requested a review from Stephen Connolly to confirm that it behaves the way he expects.</p>

<p>I found one surprise in relative path scp style URL's which is documented in the source code.  If a user wants to apply an exclusion to a relative path scp URL (like mwaite@git.markwaite.net:git/bare/bin.git), then they must prefix the relative path with a '/' so that the exclusion is applied.  The credentials Path component prefixes with a '/' if one is not detected in the string already.  That credentials code is likely used in many places, so I believe that changing that behavior would be much worse than requiring users to make this minor adaptation.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jsanchra" class="user-hover" rel="jsanchra">jsanchra</a> I revised your pull request to reduce differences to the upstream master, and to add more tests, and have submitted it as a <a href="https://github.com/jenkinsci/git-client-plugin/pull/249" class="external-link" target="_blank" rel="nofollow noopener">new pull request</a>.  It has passed my interactive testing.  I've requested a review from Stephen Connolly to confirm that it behaves the way he expects.</p>

<p>I found one surprise in relative path scp style URL's which is documented in the source code.  If a user wants to apply an exclusion to a relative path scp URL (like mwaite@​git.markwaite.net:git/bare/bin.git), then they must prefix the relative path with a '/' so that the exclusion is applied.  The credentials Path component prefixes with a '/' if one is not detected in the string already.  That credentials code is likely used in many places, so I believe that changing that behavior would be much worse than requiring users to make this minor adaptation.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=305440&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-305440">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: juan francisco sanchez
Path:
 src/main/java/org/jenkinsci/plugins/gitclient/GitURIRequirementsBuilder.java
 src/test/java/org/jenkinsci/plugins/gitclient/GitURIRequirementsBuilderTest.java
<a href="http://jenkins-ci.org/commit/git-client-plugin/a865e011e91050bc074b319f656436c3b7d88c13" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/git-client-plugin/a865e011e91050bc074b319f656436c3b7d88c13</a>
Log:
  <a href="https://issues.jenkins.io/browse/JENKINS-43022" title="Not correct exclude path to domains" class="issue-link" data-issue-key="JENKINS-43022"><del>JENKINS-43022</del></a> - Honor git credential path exclusion and inclusion</p>

<p>Tests scp URL form in GitURIRequirementsBuilderTest</p>

<p>Adds assertions for path.getPath(), including incorrect relative scp path.</p>

<p>Credentials PathRequirement incorrectly converts relative path to
absolute path.  Adapt the test to that error, since PathRequirement
is likely used in many other cases where the prefixing of a '/' to the
relative path is the expected and desired behavior.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: juan francisco sanchez<br/>
Path:<br/>
 src/main/java/org/jenkinsci/plugins/gitclient/GitURIRequirementsBuilder.java<br/>
 src/test/java/org/jenkinsci/plugins/gitclient/GitURIRequirementsBuilderTest.java<br/>
<a href="http://jenkins-ci.org/commit/git-client-plugin/a865e011e91050bc074b319f656436c3b7d88c13" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/git-client-plugin/a865e011e91050bc074b319f656436c3b7d88c13</a><br/>
Log:<br/>
  <a href="https://issue-redirect.jenkins.io/issue/43022" title="Not correct exclude path to domains" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/43022">JENKINS-43022</a>"><del>JENKINS-43022</del></a> - Honor git credential path exclusion and inclusion</p>

<p>Tests scp URL form in GitURIRequirementsBuilderTest</p>

<p>Adds assertions for path.getPath(), including incorrect relative scp path.</p>

<p>Credentials PathRequirement incorrectly converts relative path to<br/>
absolute path.  Adapt the test to that error, since PathRequirement<br/>
is likely used in many other cases where the prefixing of a '/' to the<br/>
relative path is the expected and desired behavior.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=305441&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-305441">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Mark Waite
Path:
 src/main/java/org/jenkinsci/plugins/gitclient/GitURIRequirementsBuilder.java
 src/test/java/org/jenkinsci/plugins/gitclient/GitURIRequirementsBuilderTest.java
<a href="http://jenkins-ci.org/commit/git-client-plugin/3ba074ccee0e9677e1f0431349c3e069206675dc" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/git-client-plugin/3ba074ccee0e9677e1f0431349c3e069206675dc</a>
Log:
  Merge pull request #249 from MarkEWaite/master-PR242-credentials-exclude-path</p>

<p><a href="https://issues.jenkins.io/browse/JENKINS-43022" title="Not correct exclude path to domains" class="issue-link" data-issue-key="JENKINS-43022"><del>JENKINS-43022</del></a> - Honor git credential path exclusion and inclusion</p>


<p>Compare: <a href="https://github.com/jenkinsci/git-client-plugin/compare/eef4e1d17fef...3ba074ccee0e" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-client-plugin/compare/eef4e1d17fef...3ba074ccee0e</a></p>

</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Mark Waite<br/>
Path:<br/>
 src/main/java/org/jenkinsci/plugins/gitclient/GitURIRequirementsBuilder.java<br/>
 src/test/java/org/jenkinsci/plugins/gitclient/GitURIRequirementsBuilderTest.java<br/>
<a href="http://jenkins-ci.org/commit/git-client-plugin/3ba074ccee0e9677e1f0431349c3e069206675dc" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/git-client-plugin/3ba074ccee0e9677e1f0431349c3e069206675dc</a><br/>
Log:<br/>
  Merge pull request #249 from MarkEWaite/master-PR242-credentials-exclude-path</p>

<p><a href="https://issue-redirect.jenkins.io/issue/43022" title="Not correct exclude path to domains" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/43022">JENKINS-43022</a>"><del>JENKINS-43022</del></a> - Honor git credential path exclusion and inclusion</p>


<p>Compare: <a href="https://github.com/jenkinsci/git-client-plugin/compare/eef4e1d17fef...3ba074ccee0e" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-client-plugin/compare/eef4e1d17fef...3ba074ccee0e</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=305472&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-305472">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Will be included in the next release of the git client plugin.  A pre-release build is available from the <a href="https://ci.jenkins.io/job/Plugins/job/git-client-plugin/job/master/lastBuild/" class="external-link" target="_blank" rel="nofollow noopener">last stable build of the git client plugin</a></p>
</pre>
</details>

</ul>
</details>

<p>Will be included in the next release of the git client plugin.  A pre-release build is available from the <a href="https://ci.jenkins.io/job/Plugins/job/git-client-plugin/job/master/lastBuild/" class="external-link" target="_blank" rel="nofollow noopener">last stable build of the git client plugin</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jsanchra.png" title="jsanchra's avatar" /> jsanchra:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=305495&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-305495">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> , sorry to solvent your detected issue. I've been overworked lately</p>

<p>Thanks man!!</p>
</pre>
</details>

</ul>
</details>

<p>Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> , sorry to solvent your detected issue. I've been overworked lately</p>

<p>Thanks man!!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43022?focusedId=308603&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-308603">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Git client plugin 2.5.0 released 27 Jul 2017 includes this fix</p>
</pre>
</details>

</ul>
</details>

<p>Git client plugin 2.5.0 released 27 Jul 2017 includes this fix</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a class="jira-relationship" href="https://issue-redirect.jenkins.io/issue/44773">JENKINS-44773</a>]</i>
